lib/string_validator.rb

```diff
@@ -0,0 +1,24 @@
+require 'pry'
+
+class StringValidator
+  def validate(string)
+    stack = []
+    string.split('').each do |character|
```

Comment:
It looks like you're building an array here. Can we avoid the use of the collector local variable by using a different enumerable, such as `reduce/inject`?